src/span.js

```diff
@@ -144,6 +144,17 @@ export default class Span {
      **/
     setOperationName(operationName: string): Span {
         this._operationName = operationName;
+        // If the span has debug set, then the question of resampling
+        // after setting the operation name is irrelevant.  The span is always
+        // sampled.
```

Comment:
Whenever we call finish twice, we log a message.  We can do that here.  I think this case is the same as finish because even though some implementations can support the feature there is no logical reason to.